main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -2710,20 +2840,28 @@ object Parser2 {
     private def schemaType()(implicit s: State): Mark.Closed = {
       assert(at(TokenKind.HashCurlyL))
       val mark = open()
-      separated(schemaTerm)
-        .within(TokenKind.HashCurlyL, TokenKind.CurlyR)
-        .optionallyWith(TokenKind.Bar, () => name(NAME_VARIABLE))
-        .zeroOrMore()
+      new Separated(
+        itemName = "schema term",
+        getItem = schemaTerm,
+        checkForItem = () => atAny(NAME_PREDICATE),
+        delimiters = (TokenKind.HashCurlyL, TokenKind.CurlyR),
+        recoverOn = RECOVER_TYPE,
+        optionallyWith = Some(TokenKind.Bar, () => name(NAME_VARIABLE))
+      ).zeroOrMore()
```

Comment:
Do you also construct a Separated and then _immediately_ call zeroOrMore or oneOrMore?

Reply:
If so, perhaps zeroOrMore and oneOrMore should be helper functions.

Reply:
I'm not sure I understand this. Did you mean that 
1. `zeroOrMore` should be a function that internally constructs a `Separated` and calls it or that
2. `zeroOrMore` should be a function the takes the same parameters as used for construction of `Separated` and just does the computation directly. This refactor would make us able to delete `Separated` I believe.
3. Something other than that?

Reply:
Does Separated have to be a class/object? It has no state, right?